extensions/autodoc_field/__init__.py

```diff
@@ -0,0 +1,115 @@
+from sphinx.ext.autodoc import AttributeDocumenter, ClassDocumenter
+from typing import Sequence
+
+import odoo
+import datetime
+
+
+class OdooClassDocumenter(ClassDocumenter):
+    objtype = 'odooclass'
+    directivetype = 'class'
+    priority = 10 + ClassDocumenter.priority
+
+    @classmethod
+    def can_document_member(cls, member, membername, isattr, parent):
+        return isinstance(member, odoo.models.MetaModel)
+    
+    def add_content(self, more_content):
+        sourcename = self.get_sourcename()
+        cls = self.object
+        self.add_line(f".. _model-{cls._name.replace('.', '-')}:", sourcename)
+        self.add_line('.. py:attribute:: _name' , sourcename)
+        self.add_line(f'  :value: {cls._name}', sourcename)
+        self.add_line('' , sourcename)
+        super().add_content(more_content)
+
+
+class FieldDocumenter(AttributeDocumenter):
+    objtype = 'field'
+    directivetype = 'attribute'
+    priority = 10 + AttributeDocumenter.priority
+
+    @classmethod
+    def can_document_member(cls, member, membername, isattr, parent):
+        return isinstance(member, odoo.fields.Field)
+
+    def update_annotations(self, parent):
+        super().update_annotations(parent)
+        annotation = parent.__annotations__
+        attrname = self.object.name
+        annotation[attrname] = dict
+        field = self.object
+        if field.type == 'many2one':
+            annotation[attrname] = int
+        elif field.type in ('one2many', 'many2many'):
+            annotation[attrname] = Sequence[odoo.fields.Command]
+        elif field.type in ('selection', 'reference', 'char', 'text', 'html'):
+            annotation[attrname] = str
+        elif field.type == 'boolean':
+            annotation[attrname] = bool
+        elif field.type in ('float', 'monetary'):
+            annotation[attrname] = float
+        elif field.type == 'integer':
+            annotation[attrname] = int
+        elif field.type == 'date':
+            annotation[attrname] = datetime.date
+        elif field.type == 'datetime':
+            annotation[attrname] = datetime.datetime
+
+    def add_content(self, more_content):
+        source_name = self.get_sourcename()
+        field = self.object
+        if field.required:
+            self.add_line(f"| **Required**", source_name)
+        self.add_line(f"| **Name:** {field.string}", source_name)
+        if field.readonly:
+            self.add_line(f"| **Read-only:** this field is not supposed to/cannot be set manually", source_name)
+        if not field.store:
+            self.add_line(f"| **Not stored:** this field is there only for technical reasons", source_name)
+        if field.type == 'selection':
+            if isinstance(field.selection, (list, tuple)):
+                self.add_line(f"| **Possible values:**", source_name)
+                for tech, nice in field.selection:
+                    self.add_line(f"|     `{tech}`: {nice}", source_name)
```

Comment:
```suggestion
                    self.add_line(f"|     ``{tech}``: {nice}", source_name)
```
While leveraging the default-role might be acceptable for content written up directly, it seems quite unnecessary for  generated content.